modules/aws-load-balancer-operator-install.adoc

```diff
@@ -0,0 +1,286 @@
+* Configured OpenShift (oc) CLI
+* OpenShift Container Platform (OCP) 4.13 or later
+
+[NOTE]
+====
+When installing an ALBO for use with
+ifndef::openshift-rosa[]
+{product-title}
+endif::openshift-rosa[]
+ifdef::openshift-rosa[]
+a ROSA
+endif::openshift-rosa[]
+cluster in an AWS Local Zone (LZ), the AWS LZ must be enabled for the account and AWS Elastic Load Balancing v2 (ELBv2) services must be available in the AWS LZ.
+====
+.Procedure
+
+. Identify the cluster infrastructure ID and the cluster OpenID Connect (OIDC) DNS by running the following commands:
+.. Identify the
+ifndef::openshift-rosa[]
+{product-title}
+endif::openshift-rosa[]
+ifdef::openshift-rosa[]
+ROSA
+endif::openshift-rosa[]
+cluster INFRA ID:
++
+[source,terminal]
+----
+$ rosa describe cluster --cluster=<cluster_name> | grep -i 'Infra ID'
+----
++
+or
++
+[source,terminal]
+----
+$ oc get infrastructure cluster -o json | jq -r '.status.infrastructureName'
+----
+
+.. Identify the
+ifndef::openshift-rosa[]
+{product-title}
+endif::openshift-rosa[]
+ifdef::openshift-rosa[]
+ROSA
+endif::openshift-rosa[]
+cluster OIDC DNS:
++
+--
+[source, terminal]
+----
+$ rosa describe cluster --cluster=<cluster_name> | grep -i 'OIDC'
+----
+
+Save the output from both commands. You will use this information in future steps within this procedure.
+
+--
+. Create and manage AWS Identity and Access Management (IAM) roles and policies:
++
+.. Create the AWS IAM policy required for the AWS Load Balancer Controller (ALBC) and the ALBO, which can be found link:https://github.com/openshift/aws-load-balancer-operator/blob/main/assets/iam-policy.json[here].
++
+.. Create an AWS IAM role and assign the AWS IAM policy displayed in step 2a to that role. For more information on creating AWS IAM roles, see link:https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create.html[Creating IAM roles]. For more information on adding AWS IAM policies to AWS IAM roles, see link:https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_manage-attach-detach.html[Adding and removing IAM identity permissions].
+
+.. Assign the following trust policy to the newly-created AWS IAM role. For more information on assigning trust policies to AWS IAM roles, see link:https://aws.amazon.com/blogs/security/how-to-use-trust-policies-with-iam-roles/[How to use trust policies with IAM roles].
+
++
+--
+[source,json]
+----
+"{
+    ""Version"": ""2012-10-17"",
+    ""Statement"": [
+        {
+            ""Effect"": ""Allow"",
+            ""Principal"": {
+                ""Federated"": ""arn:aws:iam::{AWS_AccountNo}: <1>
+
+                {Cluster_OIDC_Endpoint}"" <2>
+            },
+            ""Action"": ""sts:AssumeRoleWithWebIdentity"",
+            ""Condition"": {
+                ""StringEquals"": {
+                    ""{Cluster's OIDC Endpoint}:sub"": [
+                        ""system:serviceaccount:aws-load-balancer-operator:aws-load-balancer-operator-controller-manager"",
```

Comment:
> The IAM policy mentioned before is only for the controller. The operator needs its own IAM policy. It's not yet committed into `main` but it's coming, see [openshift/aws-load-balancer-operator#97](https://github.com/openshift/aws-load-balancer-operator/pull/97).
> 
> So, this part needs to be enlarged to create the roles not only for the controller but for the operator too.

Hello @alebedev87,

    Please see row 16 within the "Documentation" tab in the following spreadsheet (which I was given to use as a guide to writing these docs).  https://docs.google.com/spreadsheets/d/1nwUdersMAaKZ4AUX_KhCpgsjL6DByxVy2B95CNqaeYY/edit#gid=1241421272

Will also attach a screenshot if you don't want to deal with the link: 
![image](https://github.com/openshift/openshift-docs/assets/122639474/a5f66401-2520-4877-984c-0c05ea6895fa)
So, just to clarify, it is in your opinon that we can't coming these two policies into one at this point? Thanks.

Reply:
The IAM policy you created is not for the operator. You need the operator's IAM policy which is  is not yet available (in a form of a JSON file) on any branch of aws-load-balancer-operator repository. 

The IAM policy you used is for the controller which is created by the operator once `AWSLoadBalancerController` CR is created. And the service account for the controller will be different: `system:serviceaccount:aws-load-balancer-operator:aws-load-balancer-operator-controller-cluster`.

Reply:
Hello @alebedev87: Couple of questions here: 

1) You state "The IAM policy you created is not for the operator. You need the operator's IAM policy which is not yet available (in a form of a JSON file) on any branch of aws-load-balancer-operator repository. But soon will be _(https://github.com/openshift/aws-load-balancer-operator/pull/97)."

Reply:
The service account for the controller will be different: `system:serviceaccount:aws-load-balancer-operator:aws-load-balancer-operator-controller-cluster`.
-Updated the SA for the controller.

Reply:
> Given that, I will not be able to combine the AWS IAM policies required for the ALBC and ALBO, and the example I gave is only for the ALBC; I will need the ALBO policy separately. 

Yes.

> Looking at https://github.com/openshift/aws-load-balancer-operator/pull/97. looks to be about complete. Once it is, can you please send me the correct URL for that policy?

Ok.

> Looking at the attached screenshot, the way I understand it is that:
>
> 2a will be in reference to the AWS IAM policy for the ALBC.
>
> I will have 2b refer to the ALBO IAM policy.
> 
> Hopefully what I am asking makes sense. Thanks.

Yes, it makes sense. I would just show the ALBO IAM policy first and ALBC second because ALBO needs its credentials before ALBC. ALBC is spawned by ALBO. Here is [the link to IAM you need for ALBO](https://github.com/openshift/aws-load-balancer-operator/blob/61b55feef26e8dfd9d1b2a3300f80ba572f98d58/docs/prerequisites.md#option-2-using-aws-cli) and here is [the IAM you need for the controller](https://github.com/openshift/aws-load-balancer-operator/blob/61b55feef26e8dfd9d1b2a3300f80ba572f98d58/docs/install.md#option-2-using-the-aws-cli) (you already have but you can double check). Note the difference of the service accounts.

Reply:
Thanks. Made changes.